code/modules/power/turbine/turbine.dm

```diff
@@ -0,0 +1,557 @@
+///Multiplier for converting work done into rpm and rpm in energy out
+#define TURBINE_RPM_CONVERSION 15
+///Efficiency of the turbine to turn work into energy, higher values will yield more power
+#define TURBINE_ENERGY_RECTIFICATION_MULTIPLIER 0.25
+///Max allowed damage per tick
+#define TURBINE_MAX_TAKEN_DAMAGE 10
+///Amount of damage healed when under the heat threshold
+#define TURBINE_DAMAGE_HEALING 2
+///Amount of damage that the machine must have to start launching alarms to the engi comms
+#define TURBINE_DAMAGE_ALARM_START 30
+///Multiplier when converting the gas energy into gas work
+#define TURBINE_WORK_CONVERSION_MULTIPLIER 0.01
+///Multiplier when converting gas work back into heat
+#define TURBINE_HEAT_CONVERSION_MULTIPLIER 0.005
+///Amount of energy removed from the work done by the stator due to the consumption from the compressor working on the gases
+#define TURBINE_COMPRESSOR_STATOR_INTERACTION_MULTIPLIER 0.15
+
+/obj/machinery/power/turbine
+	density = TRUE
+	resistance_flags = FIRE_PROOF
+	can_atmos_pass = ATMOS_PASS_DENSITY
+
+	///Theoretical volume of gas that's moving through the turbine, it expands the further it goes
+	var/gas_theoretical_volume = 0
+	///Stores the turf thermal conductivity to restore it later
+	var/our_turf_thermal_conductivity
+	///Checks if the machine is processing or not
+	var/active = FALSE
+	///The parts can be registered on the main one only when their panel is closed
+	var/can_connect = TRUE
+
+	///Overlay for panel_open
+	var/open_overlay //#TODO: get the overlay done
+	///Overlay for machine activation
+	var/on_overlay //#TODO: get the overlay done
+
+	///Reference to our turbine part
+	var/obj/item/turbine_parts/installed_part
+	///Path of the turbine part we can install
+	var/part_path
+
+/obj/machinery/power/turbine/Initialize(mapload)
+	. = ..()
+
+	if(part_path)
+		installed_part = new part_path(src)
+
+	var/turf/our_turf = get_turf(src)
+	if(our_turf.thermal_conductivity != 0 && isopenturf(our_turf))
+		our_turf_thermal_conductivity = our_turf.thermal_conductivity
+		our_turf.thermal_conductivity = 0
```

Comment:
what?

Reply:
Is to prevent the floor where the turbine is to conduct heat due to superconduction since there is a hole on the wall near the middle part

Reply:
mm. I wish we had a better way of handling this

Reply:
if you have ideas lemme know and i'll make it